custom_components/localtuya/cover.py

```diff
@@ -207,87 +416,85 @@ def available(self):
-        # print('is_closing() : state [{}]'.format(state))
-        if state == "off":
+        _LOGGER.info("is_closing called")
+        last_movement = self._last_movement;
+        if last_movement == 'Closing':
             return True
         return False
 
-    @property
-    def is_closed(self):
-        """Return if the cover is closed or not."""
-        # _LOGGER.info("running is_closed from cover")
-        # self.update()
-        state = self._state
-        # print('is_closed() : state [{}]'.format(state))
-        if state == "off":
-            return False
-        if state == "on":
-            return True
-        return None
-
     def set_cover_position(self, **kwargs):
-        # _LOGGER.info("running set_cover_position from cover")
-        """Move the cover to a specific position."""
-
-        newpos = float(kwargs["position"])
-        #        _LOGGER.info("Set new pos: %f", newpos)
-
-        currpos = self.current_cover_position
-        posdiff = abs(newpos - currpos)
-        #       25 sec corrisponde alla chiusura/apertura completa
-        mydelay = posdiff / 2.0
-        if newpos > currpos:
-            #            _LOGGER.info("Opening to %f: delay %f", newpos, mydelay )
-            self.open_cover()
-        else:
-            #            _LOGGER.info("Closing to %f: delay %f", newpos, mydelay )
-            self.close_cover()
-        sleep(mydelay)
-        self.stop_cover()
-        self._position = 50  # newpos
+        """Set the cover to a specific position from 0-100"""
+        _LOGGER.debug("set_cover_position called")
+        if ATTR_POSITION in kwargs:
+            converted_position = int(kwargs[ATTR_POSITION])
+            if converted_position in range(0,101):
+                _LOGGER.info("set_cover_position about to set position to =%s", converted_position)
+                self._device.set_dps(converted_position, self._config[CONF_SET_POSITION_DPS])
+            else:
+                _LOGGER.warning("set_position given number outside range")
 
 
     def open_cover(self, **kwargs):
         """Open the cover."""
+        _LOGGER.debug("open_cover called")
         _LOGGER.debug("Launching command %s to cover ", self._config[CONF_OPEN_CMD])
         self._device.set_dps(self._config[CONF_OPEN_CMD], self._dps_id)
 
     def close_cover(self, **kwargs):
         """Close cover."""
+        _LOGGER.debug("close_cover called")
         _LOGGER.debug("Launching command %s to cover ", self._config[CONF_CLOSE_CMD])
         self._device.set_dps(self._config[CONF_CLOSE_CMD], self._dps_id)
 
     def stop_cover(self, **kwargs):
         """Stop the cover."""
-        _LOGGER.debug("Laudebugching command %s to cover ", self._config[CONF_STOP_CMD])
+        _LOGGER.debug("stop_cover called called")
+        _LOGGER.debug("Launching command %s to cover ", self._config[CONF_STOP_CMD])
         self._device.set_dps(self._config[CONF_STOP_CMD], self._dps_id)
 
     def status_updated(self):
```

Comment:
See comments about `update` earlier. You can remove this and I think you can also remove `self._state` as it doesn't seem to be used anymore.

Reply:
To clarify, what would you like me to remove here?

Reply:
You can remove this entire method and rename `update` earlier in the file to `status_updated`.